openSUSE/obs-landing
# Set page images for social media

So far we have shared three of our blog posts in Mastodon and this is how it looks, all of the posts use the same default image:

![Screenshot 2023-12-18 at 11-55-04 Open Build Service (@obshq@fosstodon org)](https://github.com/openSUSE/obs-landing/assets/2581944/5519fa27-6a5e-4187-89ea-d61a7d455d3f)

This is how we get the "page image" to be displayed on social media (using [Liquid variables and filters](https://jekyllrb.com/docs/liquid/)):
``` 
{%- assign page_image = page.image | default: '/images/obs-banner.png' | absolute_url %}
```
File [_layouts/default.html](https://github.com/openSUSE/obs-landing/blob/7f7263c8d4568e6fe08efe4c28e680a1b8e83906/_layouts/default.html#L25).

However, our blog posts always get the default image because `page.image` is never set. How can we set it in Jekyll? As simple as setting the image in the [Front Matter](https://jekyllrb.com/docs/front-matter/).

```
---
layout: post
title: The Title
category: development
image: images/posts/whatnot.png
---
```

This PR sets page images for the three posts we already have in Mastodon. They are selected from the posts and adapted to 800x400 pixels.

## Testing Tips

- Open the Deploy Preview generated by Netlify below.
- Inspect source of the page and search for `og:image` and `twitter:image`
- If they are using the path to the image you set in Front Matter, everything should be fine in social media.

## NOTE
After this is deployed, we have to:
- Edit the posts in Mastodon,
- remove the link to our blog post,
- save.
- Edit again,
- paste the same link,
- save
